public/event-emitters/event-emitter-base.js

```diff
@@ -57,6 +57,19 @@ var EventEmitter = P({
     this.listeners_by_action_key[action_type.prototype.key].push(
```

Comment:
Should this have an explicit check that `action_type.prototype.key` exists in the `listeners_by_action_key` map? I think then we'd get a nicer error when we violate expected types. As it is, I think we'd just get `Cannot call method 'push' of undefined`.